content/docs/querying/operators.md

```diff
@@ -50,31 +58,101 @@ either `0` (`false`) or `1` (`true`), depending on the comparison result.
-Vector elements for which the expression evaluates to `false` or which don't
-find an exact label match on the other side of the expression get dropped from
-the result, while the others get carried over into a result vector with their
-original (left-hand-side) metric names and data values.
+**Between two instant vectors**, these operators behave as a filter, applied to
+matching entries. Vector elements for which the expression is not true or
+which do not find a match on the other side of the expression get dropped from the
+result, while the others are propagated into a result vector with their original
+(left-hand-side) metric names and label values.
 
-# Logical/set binary operators
+### Logical/set binary operators
 
 These logical/set binary operators are only defined between instant vectors:
 
 * `and` (intersection)
-* `or` (union) (**not yet implemented!**)
+* `or` (union)
 
 `vector1 and vector2` results in a vector consisting of the elements of
 `vector1` for which there are elements in `vector2` with exactly matching
-labelsets. Other elements are dropped. The metric name and values are carried
+label sets. Other elements are dropped. The metric name and values are carried
 over from the left-hand-side vector.
 
 `vector1 or vector2` results in a vector that contains all original elements
-(labelsets + values) of `vector1` and additionally all elements of `vector2`,
-which don't have matching labelsets in `vector1`. The metric name is carried
-over from the left-hand-side vector in both cases.
+(label sets + values) of `vector1` and additionally all elements of `vector2`,
+which do not have matching label sets in `vector1`.
+
+## Vector matching
+
+Operations between vectors attempt to find a matching element in the right-hand-side
+vector for each entry in the left-hand-side. There are two basic types of
+matching behaviour:
+
+**One-to-one** finds a unique pair of entries from each side of the operation.
+In the default case, that is an operation following the format `vector1 <operator> vector2`.
+Two entries match if they have the exact same set of labels and corresponding values.
+The `on` keyword allows reducing the set of considered labels to a provided list:
+
+    <vector expr> <bin-op> on(<label list>) <vector expr>
+
+Example input:
+
+    method:http_errors:rate5m{method="get", code="500"}  24
+    method:http_errors:rate5m{method="get", code="404"}  30
+    method:http_errors:rate5m{method="put", code="500"}  3
+    method:http_errors:rate5m{method="post", code="500"} 6
+    method:http_errors:rate5m{method="post", code="404"} 21
+
+    method:http_requests:rate5m{method="get"}  600
+    method:http_requests:rate5m{method="del"}  34
+    method:http_requests:rate5m{method="post"} 120
+
+Example query:
+
+    method:http_errors:rate5m{code="500"} / on(method) method:http_requests:rate5m
+
+This returns a result vector that tells us what fraction of requests had an HTTP
```

Comment:
"tells us" -> "describes" (avoid informal language in reference docs)

Or: "This returns a result vector containing the fraction of HTTP requests with status code of 500 for each method, as measured over the last 5 minutes."